src/nodeodm_client/nodeodm_client.py

```diff
@@ -0,0 +1,237 @@
+            return
+        self.logger.info(f"Task {task_id} removed.")
+        return
+
+    def get_task_status(self, task_id):
+        """ Gets the status of a task on NodeODM.
+
+        :param task_id: The task ID to get the status for
+        :return: The status code of the task or None if an error occurred
+        """
+        try:
+            response = self.http_get(f'{self.nodeodm_url}/task/{task_id}/info')
+            response_json = json.loads(response.text)
+            if response.status_code != 200:
+                self.logger.warning(f"Error getting task status for task {task_id}: {response.text}")
+                return None
+            status_code = response_json['status']['code']
+            return status_code
+        except Exception as e:
+            self.logger.warning(f"An error occurred: {str(e)}")
+            return None
+
+    def create_task_new_init(self, options):
+        """ Creates a new task on NodeODM.
+
+        :param options: The options for the task to create
+        :return: The task ID of the created task or None if an error occurred
+        """
+        options_json = json.dumps(options)
+        files = {
+            'options': (None, options_json, 'application/json')
+        }
+        response = self.http_post(f'{self.nodeodm_url}/task/new/init', data={}, files=files)
+        response_json = json.loads(response.text)
+        if response.status_code != 200:
+            self.logger.warning(f"Error creating task: {response.text}")
+            return None
+        task_id = response_json['uuid']
+        self.logger.info(f'Creating task with UUID: {task_id} .')
+        return task_id
+
+    def task_new_commit(self, task_id):
+        """ Starts a task on NodeODM.
+
+        :param task_id: The task ID to start
+        :return: True if the task was started successfully, False if an error occurred
+        """
+        response = self.http_post(f'{self.nodeodm_url}/task/new/commit/{task_id}', data={})
+        if response.status_code != 200:
+            self.logger.warning(f"Error starting task: {response.text}")
+            return False
+        self.logger.info(f'Task with UUID: {task_id} was started successfully.')
+        return True
+
+    def task_new_upload(self, task_id, images_path):
+        """ Uploads images to a task on NodeODM. The images are uploaded in chunks of 20 images.
+        It is necessary to upload the images in chunks because otherwise python will open too many files at once.
+
+        :param task_id: The task ID to upload the images to
+        :param images_path: The path to the directory containing the images
+        :return: True if the images were uploaded successfully, False if an error occurred
+        """
+        files_to_upload = [f for f in os.listdir(images_path) if os.path.isfile(os.path.join(images_path, f))]
+
+        for i in range(0, len(files_to_upload), self.CHUNK_SIZE):
```

Comment:
Wäre es nicht zusätzlich gut, wenn die Bilder nach Channel gerechnet werden? Bzw. an welcher Stelle findet die Unterscheidung zwischen den Bildern im Channel statt? So werden ja alle Bilder aus dem Verzeichnis 1:1 in das Orthofoto geladen. Erkennt der Client automatisch, an welcher Stelle er welches Bild zur Berechnung nehmen muss?

Reply:
@saschadoemer Es wird am Ende alles in einem Orthophoto zusammengetragen. Dazu wird im Schritt Band Alignment erkannt zu welchem Band ein Bild gehört und dann anhand der features gemerged. 

Dazu versucht er möglichst viele features in Bilder wiederzuverwenden. Die Bänder können teilweise mit QGIS separat betrachtet werden. Ein Orthophoto pro Band zu berechnen würde zu größeren Problemen führen.  Beispielsweise könnte er im Band 1 andere Features finden als in Band 2 usw. Die Orthophotos würden wahrscheinlich nicht mehr vernüftig übereinander passen sobald die Abweichungen zu groß sind.